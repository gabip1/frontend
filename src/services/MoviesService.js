import axios from 'axios'

const MOVIES_API_BASE_URL = "https://backend-movie-app.herokuapp.com/api/v1/movies"

class MoviesService {

    getMovies(){
        return axios.get(MOVIES_API_BASE_URL)
    }

    createMovie(movie){
        return axios.post(MOVIES_API_BASE_URL, movie);
    }

    getMovieById(movieId){
        return axios.get(MOVIES_API_BASE_URL + '/' + movieId);
    }

    updateMovie(movie, movieId){
        return axios.put(MOVIES_API_BASE_URL + '/' + movieId, movie);
    }

    deleteMovie(movieId){
        return axios.delete(MOVIES_API_BASE_URL + '/' + movieId);
    }

}

export default new MoviesService()
