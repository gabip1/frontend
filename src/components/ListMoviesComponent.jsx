

import React, { Component } from 'react';
import MoviesService from '../services/MoviesService';

class ListMoviesComponent extends Component {
    constructor(props) {
        super(props)

        this.state = {
            movies: []
        }

        this.addMovie = this.addMovie.bind(this);
        this.editMovie = this.editMovie.bind(this);
        this.deleteMovie = this.deleteMovie.bind(this);

    }

    viewMovie(id) {
        this.props.history.push(`/view-movie/${id}`);
    }

    deleteMovie(id) {
        MoviesService.deleteMovie(id).then(res => {
            this.setState({
                movies: this.state.movies.filter(movie => movie.id !== id)
            });
        });
    }

    editMovie(id) {
        this.props.history.push(`/update-movie/${id}`);
    }

    componentDidMount() {
        MoviesService.getMovies().then((res) => {
            this.setState({ movies: res.data });
        });
    }

    addMovie() {
        this.props.history.push('/add-movie');
    }

    render() {
        return (
            <div>
                <br></br>
                <h2 className="text-center">Movies List</h2>
                <div className="row">
                    <button className="btn btn-primary" style={{ width: '150px' }} onClick={this.addMovie}>Add Movie</button>
                </div>
                <br></br>
                <div className="row">

                    <table className="table table-striped table-bordered">
                        <thead>
                            <tr>
                                <th> Movie title </th>
                                <th> Movie director </th>
                                <th> Movie genre </th>
                                <th> Actions </th>
                            </tr>
                        </thead>

                        <tbody>
                            {
                                this.state.movies.map(
                                    movie =>
                                        <tr key={movie.id}>
                                            <td> {movie.title} </td>
                                            <td> {movie.director} </td>
                                            <td> {movie.genre} </td>
                                            <td>
                                                <button onClick={() => this.viewMovie(movie.id)} className="btn btn-info">View</button>
                                                <button style={{ marginLeft: "10px" }} onClick={() => this.editMovie(movie.id)} className="btn btn-info">Edit </button>
                                                <button style={{ marginLeft: "10px" }} onClick={() => this.deleteMovie(movie.id)} className="btn btn-danger">Delete </button>
                                            </td>
                                        </tr>
                                )
                            }

                        </tbody>
                    </table>
                </div>

            </div>
        )
    }
}

export default ListMoviesComponent;
