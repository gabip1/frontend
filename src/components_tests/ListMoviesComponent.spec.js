import React from 'react';
import renderer from 'react-test-renderer';
import ListMoviesComponent from '../components/ListMoviesComponent';
import { render } from '@testing-library/react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';
import MoviesService from '../services/MoviesService';


describe('ListMoviesComponent', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<ListMoviesComponent></ListMoviesComponent>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
