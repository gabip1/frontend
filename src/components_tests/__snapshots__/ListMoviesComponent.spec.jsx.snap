// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMoviesComponent renders correctly 1`] = `
<div>
  <br />
  <h2
    className="text-center"
  >
    Movies List
  </h2>
  <div
    className="row"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      style={
        Object {
          "width": "150px",
        }
      }
    >
      Add Movie
    </button>
  </div>
  <br />
  <div
    className="row"
  >
    <table
      className="table table-striped table-bordered"
    >
      <thead>
        <tr>
          <th>
             Movie title 
          </th>
          <th>
             Movie director 
          </th>
          <th>
             Movie genre 
          </th>
          <th>
             Actions 
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
